### PR TITLE
lnav: depends_on "pcre" => "cxx11"

### LINKED ISSUE #1055: lnav: fails to install
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

_You can erase any parts of this template not applicable to your Issue._

### Bug reports:

Gist log: https://gist.github.com/fe6e35aa8a2d4b693b13adb3e19b261b

When I run `brew install lnav`, Linuxbrew fails and reports the following message:

```
Last 15 lines from /home/doc/.cache/Homebrew/Logs/lnav/02.make:
libdiag.a(readline_possibilities.o): In function `pcrecpp::RE::RE(char const*)':
readline_possibilities.cc:(.text._ZN7pcrecpp2REC2EPKc[_ZN7pcrecpp2REC5EPKc]+0x42): undefined reference to `pcrecpp::RE::In
it(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, pcrecpp::RE_Options const*)'
libdiag.a(string-extension-functions.o): In function `find_re(sqlite3_context*, char const*)':
string-extension-functions.cc:(.text+0x290): undefined reference to `pcrecpp::RE::Init(std::__cxx11::basic_string<char, st
d::char_traits<char>, std::allocator<char> > const&, pcrecpp::RE_Options const*)'
libdiag.a(string-extension-functions.o): In function `regexp_replace(sqlite3_context*, int, Mem**)':
string-extension-functions.cc:(.text+0x445): undefined reference to `pcrecpp::RE::GlobalReplace(pcrecpp::StringPiece const
&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
libdiag.a(sql_util.o): In function `attach_sqlite_db(sqlite3*, std::__cxx11::basic_string<char, std::char_traits<char>, st
d::allocator<char> > const&)':
sql_util.cc:(.text+0x3a3): undefined reference to `pcrecpp::RE::Init(std::__cxx11::basic_string<char, std::char_traits<cha
r>, std::allocator<char> > const&, pcrecpp::RE_Options const*)'
sql_util.cc:(.text+0x530): undefined reference to `pcrecpp::RE::GlobalReplace(pcrecpp::StringPiece const&, std::__cxx11::b
asic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
collect2: error: ld returned 1 exit status
Makefile:749: recipe for target 'lnav-test' failed
make[1]: *** [lnav-test] Error 1
make[1]: Leaving directory '/tmp/lnav20160407-17465-13aangc/lnav-0.8.0/src'
Makefile:463: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1
```

### PR DESCRIPTION
**Pull request by dochang**
Close #1055

### CODE REVIEW

**Comment by sjackman** [comment]
Travis fails:
https://travis-ci.org/Linuxbrew/linuxbrew/builds/122542861#L1253-L1254
```
checking for mousemask in -lncursesw... no
configure: error: libgpm development libraries are required to build
```
`libgpm` is not packaged for Linuxbrew.
See http://www.nico.schottelius.org/software/gpm/
and http://www.linuxfromscratch.org/blfs/view/stable/general/gpm.html

**Comment by dochang** [comment]
A formula for `libgpm` has been created.  Please check #1070 .

File: Library/Formula/lnav.rb [hunk 1]
```diff
@@ -19,8 +19,9 @@ class Lnav < Formula
   end
 
   depends_on "readline"
-  depends_on "pcre"
+  depends_on "pcre" => "cxx11"
   depends_on "curl" => ["with-libssh2", :optional]
+  depends_on "libgpm"
```
Comment (sjackman) [inline Library/Formula/lnav.rb]: Please change this to:
````ruby
depends_on "libgpm" unless OS.mac?
```

File: Library/Formula/lnav.rb [hunk 2]
```diff
@@ -19,8 +19,9 @@ class Lnav < Formula
   end
 
   depends_on "readline"
-  depends_on "pcre"
+  depends_on "pcre" => "cxx11"
```
Comment (sjackman) [inline Library/Formula/lnav.rb]: Please change this to
```ruby
depends_on "pcre" => [("cxx11" unless OS.mac?)].compact
```

**Comment by dochang** [comment]
The last build fails on timeout.  Please try it again.

**Comment by sjackman** [comment]
I'm currently splitting Linuxbrew/linuxbrew into Linuxbrew/brew and Linuxbrew/homebrew-core, as Homebrew did last week (two weeks ago?), so I'm not merging any pull requests right now. Sorry for the delay.

**Comment by sjackman** [comment]
Rebuilding on Travis.

**Comment by sjackman** [comment]
Moved to https://github.com/Linuxbrew/homebrew-core/pull/1848

**Pull request closed without merging**